Python/ceval.c

```diff
@@ -2024,8 +2024,8 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
             STAT_INC(BINARY_OP, hit);
             PyObject *prod = _PyLong_Multiply((PyLongObject *)left, (PyLongObject *)right);
             SET_SECOND(prod);
-            Py_DECREF(right);
-            Py_DECREF(left);
+            _Py_DECREF_SPECIALIZED(right, (destructor)_PyLong_ExactDealloc);
```

Comment:
There are a lot of casts. Could you make `_PyLong_ExactDealloc` and `_PyFloat_ExactDealloc` take `PyObject *`?